src/asmjit/core/jitallocator.h

```diff
@@ -54,6 +54,10 @@ enum class JitAllocatorOptions : uint32_t {
   //! or have all blocks fully occupied.
   kImmediateRelease = 0x00000008u,
 
+  //! Enables the use of large pages for virtual memory allocations to improve iTLB efficiency.  For example, on a
+  //! 64-bit x86, virtual memory allocations might use 2MiB large pages, instead of the 4KiB default pages.
+  kUseLargePages = 0x00000010u,
```

Comment:
I'm wondering - if we fail to allocate a large page - do we fallback to standard pages?

Reply:
Yeah, that is a good call-out.  So, I think that this mostly happens today with two exceptions

1) On Windows, `VirtualAlloc` with the `MEM_LARGE_PAGES` flag will fail if it is unable to allocate large pages
2) On Linux, the `memfd_create` call with the `MFD_HUGEPAGE | MFD_HUGE_2MB` flag will fail if there are no HugeTLB pages available

I think in both cases we can check for an error and invoke the normal code-paths.  What do you think?

Reply:
I think this is a little more complicated than I had assumed on Linux  when using memfd.  There, when large pages are exhausted there is no way to fallback to allocating default pages using the same file descriptor.

I think we can work around this by either

1) documenting the limitation
2) creating two file descriptors with `memfd_create(2)` and using one to allocate as many large pages as possible and when that is exhausted, allocate regular pages through the other file descriptor

What do you think?

Reply:
@kobalicek let me know if this what you had in mind as a fallback or if a fallback in this case is needed so i can get the diff into the right end-state.  Thanks!

Reply:
I think it sounds good. I personally don't know any details regarding closing the anonymous FDs. I though that once you allocate dual-mapping, for example, you can just close the descriptors and be done with that. I mean no leak once you unmap the pages basically.

I don't mind opening memfd twice if we fail to allocate huge pages. It's a border case, it doesn't need optimizations I think.